codegenerator/cli/templates/static/codegen/src/eventFetching/FetchState.res

```diff
@@ -188,112 +144,70 @@ let shallowCopyRegister = (register: register) => {
 }
 
 let copy = (self: t) => {
-  let rec loop = (register: register, ~parent=?) =>
-    switch register.registerType {
-    | RootRegister =>
-      let copied = register->shallowCopyRegister
-      switch parent {
-      | Some(parent) => parent->Parent.joinChild(copied)
-      | None => copied
-      }
-    | DynamicContractRegister({id, nextRegister}) =>
-      nextRegister->loop(
-        ~parent=register->shallowCopyRegister->Parent.make(~dynamicContractId=id, ~parent),
-      )
-    }
-
-  let baseRegister = loop(self.baseRegister)
   let pendingDynamicContracts = self.pendingDynamicContracts->Array.copy
+  let registers = self.registers->Array.map(shallowCopyRegister)
   {
     partitionId: self.partitionId,
-    baseRegister,
+    registers,
+    // Must use the reference to copied value, so we use find
+    mostBehindRegister: registers
+    ->Js.Array2.find(r => r.id == self.mostBehindRegister.id)
+    ->Option.getExn,
+    nextMostBehindRegister: switch self.nextMostBehindRegister {
+    | Some(nextMostBehindRegister) =>
+      registers->Js.Array2.find(r => r.id == nextMostBehindRegister.id)->Option.getExn->Some
+    | None => None
+    },
```

Comment:
Just a thought, maybe worth using an array index here rather than an obj reference? Not sure of the performance hit though.

Looks something like an arena/index tree data structure.

Reply:
I think a reference should be faster, since we don't need to get it every time we need it. Regarding memory, it should be the same. In any way the difference is neglectable.

Also, `find` looks dirty here, but I've decided to keep it as it is, since I was thinking to get rid of `mostBehindRegister` and `nextMostBehindRegister` when I turn registers into partitions.